File: Trainer/OTraining.Core/Models/OTrainingDto.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace OTraining.Core.Models
{
    public class OTrainingDto
    {
        public OTrainingDto()
        {
            ProgramsDto = new HashSet<OTrainingProgramDto>();
            DetailsDto = new OTrainingDetailsDto();
        }
        public OTrainingDetailsDto DetailsDto { get; set; }
        public IEnumerable<OTrainingProgramDto> ProgramsDto { get; set; }
    }
}

File: Trainer-FrontEnd/src/app/shared/models/otraining/otraining-dto.ts
export class OTrainingDto {
    detailsDto: OTrainingDetailsDto;
    programsDto: OTrainingProgramDto[];
}

export class OTrainingDetailsDto {
    description: string = "";
    forJoin: string = "";
}

export class OTrainingProgramDto {
    id: number = 0;
    name: string = "";
    features: string = "";
    profilePicture: string ="";
    profilePictureFile: File;
}